src/app/mod.rs

```diff
@@ -77,19 +77,18 @@ pub struct App<'app> {
 }
 
 impl<'app> App<'app> {
-    pub fn new(state: state::AppState<'app>) -> Self {
+    pub fn new(state: state::AppState<'app>, execution: ExecutionContext) -> Self {
```

Comment:
I moved the ExecutionContext creation out of the application state (as it can now potentially return an Error)